src/coreComponents/managers/Functions/TableFunction.hpp

```diff
@@ -136,14 +142,19 @@ class TableFunction : public FunctionBase
    * @brief Set the table coordinates
    * @param coordinates An array of arrays containing table coordinate definitions
    */
-  void setTableCoordinates( array1d< real64_array > coordinates ) { m_coordinates = coordinates; }
+  void setTableCoordinates( ArrayOfArrays< real64 > coordinates ) { m_coordinates = coordinates; }
```

Comment:
This wasn't right in old version either, but we should avoid double-copying
```suggestion
  void setTableCoordinates( ArrayOfArrays< real64 > coordinates ) { m_coordinates = std::move( coordinates ); }
```

I think though that perhaps the interface should be left as it was (i.e. passing an `array1d< array1d< real64 > >` for user convenience) and internally build the `ArrayOfArrays` in this function.

Also it may be reasonable to call `reInitializeFunction()` inside all of these `set...` methods so as to keep the object always internally consistent.

Reply:
I agree. I went back to the original interface, passing just an `array1d< array1d< real64 > >` and then constructing internally the `ArrayOfArrays`. Also, now I call the `reInitializeFunction` inside all the setters to make sure that the table wrapper is always consistent.